docs/extensions/code-review-tool-attestations.md

```diff
@@ -0,0 +1,114 @@
+# Code Review Tool Attestations
+
+Last Modified: October 22, 2024
+
+gittuf's [reference authorization
+attestations](/docs/design-document.md#reference-authorization) can be used to
+meet a threshold of approvals for a change in a Git repository. Here, every
+approver signs the attestation themselves, minimizing the trusted entities
+during verification, but this has usability concerns. To balance usability and
+security, this extension proposes the use of special attestations created by
+popular code review tools.
+
+## Attestation Schema
+
+The attestation is similar to the reference authorization, but also includes
+information indicating the approvers.
+
+```
+TargetRef    string
+FromTargetID string
+ToTargetID   string
+Approvers    []string
+```
+
+The attestation must indicate the code review tool via the predicate type, and
+it must be signed by the code review tool or a trusted integration with the
+tool. See [verification](#verification) for how these attestations are
+authenticated.
+
+## Policy
+
+gittuf policy must be updated to extend the information tracked for each trusted
+person. Currently, each trusted person is represented directly by their public
+key in policy metadata, similar to how TUF represents actors. To support code
+review tool attestations, this must be extended to support the identity of an
+approver from the perspective of the code review tool. For example, Alice may
+use an SSH key for commits and RSL entries but use GitHub's cloud offering for
+code review. Thus, gittuf policy must associate her public key and her GitHub
+username / identifier. Also see: https://github.com/gittuf/gittuf/issues/586.
+
+This extension proposes the following format for the policy schema:
```

Comment:
I think I get what you're getting at, but could you rephrase to be certain?

Reply:
Good points, and I'll raise you one!

Right now, I'm focusing on allowing "alice" to sign using any of her keys. I think we could explore parametrizing it in future, like `alice:<keyid>` or something, but I'm unsure how frequently this is going to be necessary. I also see no reason we don't continue supporting `Key` as a type where this is a concern. But overall, I think the point you raise is an open question: do we support setting constraints on a person as to what keys they ought to use in a particular context? What about thresholds? Right now, it only counts towards the threshold for the rule but perhaps in a context, Alice can only count as having signed if she uses _two of her keys_ rather than just one.

The other thing I was considering: consistency in Person definitions. Is a `Person` redefined in multiple rule files where they're delegated to? How do we keep them consistent across definitions?

Reply:
This is really a two-parter:
a) Storing metadata alongside a person definition pertaining to their github etc identities
b) Using those identities to verify code review attestations from SCPs